cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/Context/SchemaGenerator.ts

```diff
@@ -0,0 +1,240 @@
+    }
+  }
+  return nullable ? [finalType, 'null'] : finalType;
+}
+
+function generateMapType(children: IOrderedChildren, nullable) {
+  const finalType = {
+    type: 'map',
+    keys: 'string',
+    values: 'string',
+  };
+  for (const childId of Object.keys(children)) {
+    const currentChild = children[childId];
+    const { type, nullable: isCurrentChildNullable, internalType } = currentChild;
+    const isMapChildComplexType = isDisplayTypeComplex({ type });
+    if (!isMapChildComplexType) {
+      if (internalType === InternalTypesEnum.MAP_KEYS_SIMPLE_TYPE) {
+        finalType.keys = isCurrentChildNullable ? [type, 'null'] : type;
+      }
+      if (internalType === InternalTypesEnum.MAP_VALUES_SIMPLE_TYPE) {
+        finalType.values = isCurrentChildNullable ? [type, 'null'] : type;
+      }
+      continue;
+    }
+    // nested complex types.
+    const complexType = generateSchemaFromComplexType(type, currentChild, isCurrentChildNullable);
+    if (internalType === InternalTypesEnum.MAP_KEYS_COMPLEX_TYPE_ROOT) {
+      finalType.keys = complexType as any;
+    }
+    if (internalType === InternalTypesEnum.MAP_VALUES_COMPLEX_TYPE_ROOT) {
+      finalType.values = complexType as any;
+    }
+  }
+  return nullable ? [finalType, 'null'] : finalType;
+}
+
+function generateEnumType(children: IOrderedChildren, currentNode: INode, nullable) {
+  const finalType: IEnumFieldBase = {
+    type: 'enum',
+    symbols: [],
+  };
+  const { typeProperties: currentTypeProperties = {} } = currentNode;
+  if (currentTypeProperties.doc) {
+    finalType.doc = currentTypeProperties.doc;
+  }
+  if (currentTypeProperties.aliases) {
+    finalType.aliases = currentTypeProperties.aliases;
+  }
+  if (Array.isArray(children.order)) {
+    for (const childId of children.order) {
+      const currentChild = children[childId];
+      const { typeProperties } = currentChild;
+      if (typeProperties.symbol && typeProperties.symbol !== '') {
+        finalType.symbols.push(typeProperties.symbol);
+      }
+      if (typeProperties.doc) {
+        finalType.doc = typeProperties.doc;
+      }
+      if (typeProperties.aliases) {
+        finalType.aliases = typeProperties.aliases;
+      }
+    }
+  }
+  return nullable ? [finalType, 'null'] : finalType;
+}
+
+function generateRecordType(children: IOrderedChildren, currentNode: INode, nullable: boolean) {
+  const finalType: IRecordField = {
+    type: 'record',
+    name: currentNode.name || `name-${uuidV4()}`,
+    fields: [],
+  };
+  const { typeProperties = {} } = currentNode;
+  if (typeProperties.doc) {
+    finalType.doc = typeProperties.doc;
+  }
+  if (typeProperties.aliases) {
+    finalType.aliases = typeProperties.aliases;
+  }
+  if (Array.isArray(children.order)) {
+    for (const childId of children.order) {
+      const currentChild = children[childId];
+      const { name, type, nullable: isFiledNullable } = currentChild;
+      if (!name || name === '') {
```

Comment:
I believe `if (!name)` will suffice since this will check for empty string as well.